components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Discovery/src/Services/DiscoveryServices.cs

```diff
@@ -281,7 +281,7 @@ public class DiscoveryServices : IDiscoveryServices, IScannerServices, IDisposab
             _logger.Information("Start {mode} discovery run...", request.Mode);
             var watch = Stopwatch.StartNew();
 
-            if (request.Mode == DiscoveryMode.Url) {
+            if (request.Mode == DiscoveryMode.Url || request.Mode == DiscoveryMode.Off) {
```

Comment:
Is this why in master discovery did not work anymore?

Reply:
The real fix is the change in the OnboardingClient.cs, and this change is a safeguard.

Reply:
Why do we discover when DiscoveryMode is Off?

Reply:
It's the case when discovery request has DiscoveryMode.Off set and contains discoveryUrl(s). This was supposed to go targeted on the provided url(s) and get all endpoints, but not do any network scanning. The DiscoveryMode.Url was introduced later and, with this, the semantic of DiscoveryMode.Off was changed (in a pretty dramatic way - Off was actually triggering a full blown network scan that). I just restored the old semantic for backwards compatibility. We can discuss if we want to change the semantic here, but for now I just fixed the bug.